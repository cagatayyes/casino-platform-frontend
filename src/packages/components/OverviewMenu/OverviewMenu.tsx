
import { JSX } from 'react';
import Navigation from 'casino/components/Navigation/Navigation';
import { OverviewMenuItems } from 'casino/components/Navigation/Navigation.constants';
import { INavigationTriggerProps } from 'casino/components/Navigation/Navigation.type';
import { RecentlyPlayedCategory } from 'casino/components/CategoryList/RecentlyPlayedCategory/RecentlyPlayedCategory';
import { useTranslation } from 'react-i18next';
import { useToken } from 'casino/hooks';

const OverviewMenu = (props: INavigationTriggerProps): JSX.Element => {
	const { t } = useTranslation();
	const { userToken } = useToken();

	return (
		<div className='ml-[10px]'>
			<Navigation
				options={OverviewMenuItems}
				classNames='flex flex-row w-full items-left items-start cursor-pointer p-[6px] '
				handleTabClick={props.handleTabClick}
				activeTab={props.activeTab}
			/>
			{userToken && <RecentlyPlayedCategory
				title={t('continuePlaying')}
				showViewAllButton={false}
				showCategoryIcon={false}
				showLoading={true}
			/>}

		</div>
	);
};

export default OverviewMenu;

