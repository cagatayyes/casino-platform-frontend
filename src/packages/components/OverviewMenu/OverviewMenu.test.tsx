/* eslint-disable react/prop-types */
/* eslint-disable @typescript-eslint/explicit-function-return-type */
import { render } from '@testing-library/react';
import OverviewMenu from './OverviewMenu';
import React from 'react';

jest.mock('swiper/react', () => ({
	Swiper: ({ children }): JSX.Element => <div>{children}</div>,
	SwiperSlide: ({ children }): JSX.Element => <div>{children}</div>,
}));

jest.mock('swiper/modules', () => ({
	Swiper: ({ children }): JSX.Element => <div>{children}</div>,
	SwiperSlide: ({ children }): JSX.Element => <div>{children}</div>,
}));

jest.mock('casino/hooks/useDatePattern', () => ({
	useDatePattern: () => ({
		datePattern: '1708591412' // Mocked date pattern
	})
}));

jest.mock('casino/hooks/useToken', () => ({
	useToken: () => ({
		token: '',
		userToken: false,
	})
}));

describe('Overview Information', () => {
	it('should render Overview Information details', () => {
		const component = render(
			<OverviewMenu
				handleTabClick={jest.fn()}
				maxWidth='640px'
				activeTab='Customer Support'
			/>,
		);
		expect(component).toMatchSnapshot();
	});
});
