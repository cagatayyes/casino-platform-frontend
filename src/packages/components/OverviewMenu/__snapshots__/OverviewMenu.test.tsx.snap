// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview Information should render Overview Information details 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ml-[10px]"
      >
        <div
          class="flex flex-row w-full items-left items-start cursor-pointer p-[6px]  "
        >
          <span
            class="w-[20] h-[20] material-symbols-outlined"
            style="font-size: 20px;"
          >
            account_circle
          </span>
          <span
            class="text-[12px] font-semibold pl-[5px]"
          >
            Account
          </span>
          <span
            class="absolute right-[16px] mt-[5px] justify-right material-symbols-outlined"
            style="font-size: 11px;"
          >
            arrow_right_alt
          </span>
        </div>
        <hr
          class="w-full border-b-[1px] broder-b-[#ffffff] opacity-[0.1] m-[5px]"
        />
        <div
          class="flex flex-row w-full items-left items-start cursor-pointer p-[6px]  "
        >
          <span
            class="w-[20] h-[20] material-symbols-outlined"
            style="font-size: 20px;"
          >
            outbound
          </span>
          <span
            class="text-[12px] font-semibold pl-[5px]"
          >
            Withdraw
          </span>
          <span
            class="absolute right-[16px] mt-[5px] justify-right material-symbols-outlined"
            style="font-size: 11px;"
          >
            arrow_right_alt
          </span>
        </div>
        <hr
          class="w-full border-b-[1px] broder-b-[#ffffff] opacity-[0.1] m-[5px]"
        />
        <div
          class="flex flex-row w-full items-left items-start cursor-pointer p-[6px]  "
        >
          <span
            class="w-[20] h-[20] material-symbols-outlined"
            style="font-size: 20px;"
          >
            add_to_photos
          </span>
          <span
            class="text-[12px] font-semibold pl-[5px]"
          >
            Deposit
          </span>
          <span
            class="absolute right-[16px] mt-[5px] justify-right material-symbols-outlined"
            style="font-size: 11px;"
          >
            arrow_right_alt
          </span>
        </div>
        <hr
          class="w-full border-b-[1px] broder-b-[#ffffff] opacity-[0.1] m-[5px]"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ml-[10px]"
    >
      <div
        class="flex flex-row w-full items-left items-start cursor-pointer p-[6px]  "
      >
        <span
          class="w-[20] h-[20] material-symbols-outlined"
          style="font-size: 20px;"
        >
          account_circle
        </span>
        <span
          class="text-[12px] font-semibold pl-[5px]"
        >
          Account
        </span>
        <span
          class="absolute right-[16px] mt-[5px] justify-right material-symbols-outlined"
          style="font-size: 11px;"
        >
          arrow_right_alt
        </span>
      </div>
      <hr
        class="w-full border-b-[1px] broder-b-[#ffffff] opacity-[0.1] m-[5px]"
      />
      <div
        class="flex flex-row w-full items-left items-start cursor-pointer p-[6px]  "
      >
        <span
          class="w-[20] h-[20] material-symbols-outlined"
          style="font-size: 20px;"
        >
          outbound
        </span>
        <span
          class="text-[12px] font-semibold pl-[5px]"
        >
          Withdraw
        </span>
        <span
          class="absolute right-[16px] mt-[5px] justify-right material-symbols-outlined"
          style="font-size: 11px;"
        >
          arrow_right_alt
        </span>
      </div>
      <hr
        class="w-full border-b-[1px] broder-b-[#ffffff] opacity-[0.1] m-[5px]"
      />
      <div
        class="flex flex-row w-full items-left items-start cursor-pointer p-[6px]  "
      >
        <span
          class="w-[20] h-[20] material-symbols-outlined"
          style="font-size: 20px;"
        >
          add_to_photos
        </span>
        <span
          class="text-[12px] font-semibold pl-[5px]"
        >
          Deposit
        </span>
        <span
          class="absolute right-[16px] mt-[5px] justify-right material-symbols-outlined"
          style="font-size: 11px;"
        >
          arrow_right_alt
        </span>
      </div>
      <hr
        class="w-full border-b-[1px] broder-b-[#ffffff] opacity-[0.1] m-[5px]"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
